diff --git a/app/utils/email.server.ts b/app/utils/email.server.ts
index f62a369..5ca9264 100644
--- a/app/utils/email.server.ts
+++ b/app/utils/email.server.ts
@@ -1,4 +1,4 @@
-import { renderAsync } from '@react-email/components'
+import { render } from '@react-email/components'
 import { type ReactElement } from 'react'
 import { z } from 'zod'
 
@@ -91,8 +91,8 @@ export async function sendEmail({
 
 async function renderReactEmail(react: ReactElement) {
 	const [html, text] = await Promise.all([
-		renderAsync(react),
-		renderAsync(react, { plainText: true }),
+		render(react),
+		render(react, { plainText: true }),
 	])
 	return { html, text }
 }
