diff --git a/app/utils/verification.server.ts b/app/utils/verification.server.ts
index 6cfc4a8..1099f7c 100644
--- a/app/utils/verification.server.ts
+++ b/app/utils/verification.server.ts
@@ -1,4 +1,4 @@
-import { createCookieSessionStorage } from '@remix-run/node'
+import { createCookieSessionStorage } from 'react-router'
 
 export const verifySessionStorage = createCookieSessionStorage({
 	cookie: {
