diff --git a/app/routes/_seo+/robots[.]txt.ts b/app/routes/_seo+/robots[.]txt.ts
index 94280a1..4622d7b 100644
--- a/app/routes/_seo+/robots[.]txt.ts
+++ b/app/routes/_seo+/robots[.]txt.ts
@@ -1,5 +1,5 @@
 import { generateRobotsTxt } from '@nasa-gcn/remix-seo'
-import { type LoaderFunctionArgs } from '@remix-run/node'
+import { type LoaderFunctionArgs } from 'react-router'
 import { getDomainUrl } from '#app/utils/misc.tsx'
 
 export function loader({ request }: LoaderFunctionArgs) {
