diff --git a/app/routes/users+/index.tsx b/app/routes/users+/index.tsx
index c070332..338d7fd 100644
--- a/app/routes/users+/index.tsx
+++ b/app/routes/users+/index.tsx
@@ -1,5 +1,10 @@
-import { json, redirect, type LoaderFunctionArgs } from '@remix-run/node'
-import { Link, useLoaderData } from '@remix-run/react'
+import {
+	data,
+	redirect,
+	type LoaderFunctionArgs,
+	Link,
+	useLoaderData,
+} from 'react-router'
 import { z } from 'zod'
 import { GeneralErrorBoundary } from '#app/components/error-boundary.tsx'
 import { ErrorList } from '#app/components/forms.tsx'
@@ -41,11 +46,11 @@ export async function loader({ request }: LoaderFunctionArgs) {
 
 	const result = UserSearchResultsSchema.safeParse(rawUsers)
 	if (!result.success) {
-		return json({ status: 'error', error: result.error.message } as const, {
+		return data({ status: 'error', error: result.error.message } as const, {
 			status: 400,
 		})
 	}
-	return json({ status: 'idle', users: result.data } as const)
+	return { status: 'idle', users: result.data } as const
 }
 
 export default function UsersRoute() {
