diff --git a/app/utils/toast.server.ts b/app/utils/toast.server.ts
index d25b064..b46fefa 100644
--- a/app/utils/toast.server.ts
+++ b/app/utils/toast.server.ts
@@ -1,5 +1,5 @@
 import { createId as cuid } from '@paralleldrive/cuid2'
-import { createCookieSessionStorage, redirect } from '@remix-run/node'
+import { createCookieSessionStorage, redirect } from 'react-router'
 import { z } from 'zod'
 import { combineHeaders } from './misc.tsx'
 
