diff --git a/app/components/error-boundary.tsx b/app/components/error-boundary.tsx
index 573b412..262851b 100644
--- a/app/components/error-boundary.tsx
+++ b/app/components/error-boundary.tsx
@@ -1,16 +1,17 @@
+import { captureException } from '@sentry/react'
+import { useEffect, type ReactElement } from 'react'
 import {
 	type ErrorResponse,
 	isRouteErrorResponse,
 	useParams,
 	useRouteError,
-} from '@remix-run/react'
-import { captureRemixErrorBoundaryError } from '@sentry/remix'
-import { getErrorMessage } from '#app/utils/misc.tsx'
+} from 'react-router'
+import { getErrorMessage } from '#app/utils/misc'
 
 type StatusHandler = (info: {
 	error: ErrorResponse
 	params: Record<string, string | undefined>
-}) => JSX.Element | null
+}) => ReactElement | null
 
 export function GeneralErrorBoundary({
 	defaultStatusHandler = ({ error }) => (
@@ -23,16 +24,19 @@ export function GeneralErrorBoundary({
 }: {
 	defaultStatusHandler?: StatusHandler
 	statusHandlers?: Record<number, StatusHandler>
-	unexpectedErrorHandler?: (error: unknown) => JSX.Element | null
+	unexpectedErrorHandler?: (error: unknown) => ReactElement | null
 }) {
 	const error = useRouteError()
-	captureRemixErrorBoundaryError(error)
 	const params = useParams()
 
 	if (typeof document !== 'undefined') {
 		console.error(error)
 	}
 
+	useEffect(() => {
+		captureException(error)
+	}, [error])
+
 	return (
 		<div className="container flex items-center justify-center p-20 text-h2">
 			{isRouteErrorResponse(error)
