diff --git a/app/routes/resources+/theme-switch.tsx b/app/routes/resources+/theme-switch.tsx
index 5ea2aea..37337a9 100644
--- a/app/routes/resources+/theme-switch.tsx
+++ b/app/routes/resources+/theme-switch.tsx
@@ -1,13 +1,21 @@
 import { useForm, getFormProps } from '@conform-to/react'
 import { parseWithZod } from '@conform-to/zod'
 import { invariantResponse } from '@epic-web/invariant'
-import { json, type ActionFunctionArgs } from '@remix-run/node'
-import { redirect, useFetcher, useFetchers } from '@remix-run/react'
+import {
+	data,
+	type ActionFunctionArgs,
+	redirect,
+	useFetcher,
+	useFetchers,
+} from 'react-router'
 import { ServerOnly } from 'remix-utils/server-only'
 import { z } from 'zod'
 import { Icon } from '#app/components/ui/icon.tsx'
-import { useHints } from '#app/utils/client-hints.tsx'
-import { useRequestInfo } from '#app/utils/request-info.ts'
+import { useHints, useOptionalHints } from '#app/utils/client-hints.tsx'
+import {
+	useOptionalRequestInfo,
+	useRequestInfo,
+} from '#app/utils/request-info.ts'
 import { type Theme, setTheme } from '#app/utils/theme.server.ts'
 
 const ThemeFormSchema = z.object({
@@ -32,7 +40,7 @@ export async function action({ request }: ActionFunctionArgs) {
 	if (redirectTo) {
 		return redirect(redirectTo, responseInit)
 	} else {
-		return json({ result: submission.reply() }, responseInit)
+		return data({ result: submission.reply() }, responseInit)
 	}
 }
 
@@ -129,3 +137,13 @@ export function useTheme() {
 	}
 	return requestInfo.userPrefs.theme ?? hints.theme
 }
+
+export function useOptionalTheme() {
+	const optionalHints = useOptionalHints()
+	const optionalRequestInfo = useOptionalRequestInfo()
+	const optimisticMode = useOptimisticThemeMode()
+	if (optimisticMode) {
+		return optimisticMode === 'system' ? optionalHints?.theme : optimisticMode
+	}
+	return optionalRequestInfo?.userPrefs.theme ?? optionalHints?.theme
+}
