diff --git a/app/routes/admin+/cache_.lru.$cacheKey.ts b/app/routes/admin+/cache_.lru.$cacheKey.ts
index 2b793d0..792c929 100644
--- a/app/routes/admin+/cache_.lru.$cacheKey.ts
+++ b/app/routes/admin+/cache_.lru.$cacheKey.ts
@@ -1,5 +1,5 @@
 import { invariantResponse } from '@epic-web/invariant'
-import { json, type LoaderFunctionArgs } from '@remix-run/node'
+import { type LoaderFunctionArgs } from 'react-router'
 import { lruCache } from '#app/utils/cache.server.ts'
 import {
 	getAllInstances,
@@ -19,7 +19,7 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
 
 	const { cacheKey } = params
 	invariantResponse(cacheKey, 'cacheKey is required')
-	return json({
+	return {
 		instance: {
 			hostname: instance,
 			region: allInstances[instance],
@@ -27,5 +27,5 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
 		},
 		cacheKey,
 		value: lruCache.get(cacheKey),
-	})
+	}
 }
