diff --git a/app/routes/$.tsx b/app/routes/$.tsx
index b26d52a..52c33e3 100644
--- a/app/routes/$.tsx
+++ b/app/routes/$.tsx
@@ -5,11 +5,15 @@
 // ensure the user gets the right status code and we can display a nicer error
 // message for them than the Remix and/or browser default.
 
-import { Link, useLocation } from '@remix-run/react'
+import { Link, useLocation } from 'react-router'
 import { GeneralErrorBoundary } from '#app/components/error-boundary.tsx'
 import { Icon } from '#app/components/ui/icon.tsx'
 
-export async function loader() {
+export function loader() {
+	throw new Response('Not found', { status: 404 })
+}
+
+export function action() {
 	throw new Response('Not found', { status: 404 })
 }
 
