diff --git a/app/utils/client-hints.tsx b/app/utils/client-hints.tsx
index 546dfd2..baf72d7 100644
--- a/app/utils/client-hints.tsx
+++ b/app/utils/client-hints.tsx
@@ -8,9 +8,9 @@ import {
 	subscribeToSchemeChange,
 } from '@epic-web/client-hints/color-scheme'
 import { clientHint as timeZoneHint } from '@epic-web/client-hints/time-zone'
-import { useRevalidator } from '@remix-run/react'
+import { useRevalidator } from 'react-router'
 import * as React from 'react'
-import { useRequestInfo } from './request-info.ts'
+import { useOptionalRequestInfo, useRequestInfo } from './request-info.ts'
 
 const hintsUtils = getHintUtils({
 	theme: colorSchemeHint,
@@ -28,6 +28,11 @@ export function useHints() {
 	return requestInfo.hints
 }
 
+export function useOptionalHints() {
+	const requestInfo = useOptionalRequestInfo()
+	return requestInfo?.hints
+}
+
 /**
  * @returns inline script element that checks for client hints and sets cookies
  * if they are not set then reloads the page if any cookie was set to an
