diff --git a/app/utils/db.server.ts b/app/utils/db.server.ts
index 9e010e8..1d3c444 100644
--- a/app/utils/db.server.ts
+++ b/app/utils/db.server.ts
@@ -1,5 +1,6 @@
 import { remember } from '@epic-web/remember'
-import { PrismaClient } from '@prisma/client'
+// Changed import due to issue: https://github.com/remix-run/react-router/pull/12644
+import { PrismaClient } from '@prisma/client/index.js'
 import chalk from 'chalk'
 
 export const prisma = remember('prisma', () => {
