diff --git a/app/routes/settings+/profile.change-email.server.tsx b/app/routes/settings+/profile.change-email.server.tsx
index 0a0eebc..2c31ac4 100644
--- a/app/routes/settings+/profile.change-email.server.tsx
+++ b/app/routes/settings+/profile.change-email.server.tsx
@@ -1,6 +1,6 @@
 import { invariant } from '@epic-web/invariant'
 import * as E from '@react-email/components'
-import { json } from '@remix-run/node'
+import { data } from 'react-router'
 import {
 	requireRecentVerification,
 	type VerifyFunctionArgs,
@@ -26,7 +26,7 @@ export async function handleVerification({
 	)
 	const newEmail = verifySession.get(newEmailAddressSessionKey)
 	if (!newEmail) {
-		return json(
+		return data(
 			{
 				result: submission.reply({
 					formErrors: [
