diff --git a/app/utils/misc.tsx b/app/utils/misc.tsx
index cd8e7e1..e3ad6c6 100644
--- a/app/utils/misc.tsx
+++ b/app/utils/misc.tsx
@@ -1,4 +1,4 @@
-import { useFormAction, useNavigation } from '@remix-run/react'
+import { useFormAction, useNavigation } from 'react-router'
 import { clsx, type ClassValue } from 'clsx'
 import { useEffect, useMemo, useRef, useState } from 'react'
 import { useSpinDelay } from 'spin-delay'
