diff --git a/app/routes/users+/$username.test.tsx b/app/routes/users+/$username.test.tsx
index c9d32dc..01c3359 100644
--- a/app/routes/users+/$username.test.tsx
+++ b/app/routes/users+/$username.test.tsx
@@ -2,8 +2,8 @@
  * @vitest-environment jsdom
  */
 import { faker } from '@faker-js/faker'
-import { createRemixStub } from '@remix-run/testing'
 import { render, screen } from '@testing-library/react'
+import { createRoutesStub } from 'react-router'
 import setCookieParser from 'set-cookie-parser'
 import { test } from 'vitest'
 import { loader as rootLoader } from '#app/root.tsx'
@@ -21,7 +21,7 @@ test('The user profile when not logged in as self', async () => {
 		select: { id: true, username: true, name: true },
 		data: { ...createUser(), image: { create: userImage } },
 	})
-	const App = createRemixStub([
+	const App = createRoutesStub([
 		{
 			path: '/users/:username',
 			Component: UsernameRoute,
@@ -61,7 +61,7 @@ test('The user profile when logged in as self', async () => {
 		[parsedCookie.name]: parsedCookie.value,
 	}).toString()
 
-	const App = createRemixStub([
+	const App = createRoutesStub([
 		{
 			id: 'root',
 			path: '/',
