diff --git a/app/routes/users+/$username_+/notes.new.tsx b/app/routes/users+/$username_+/notes.new.tsx
index ecc8580..3de17d8 100644
--- a/app/routes/users+/$username_+/notes.new.tsx
+++ b/app/routes/users+/$username_+/notes.new.tsx
@@ -1,4 +1,4 @@
-import { json, type LoaderFunctionArgs } from '@remix-run/node'
+import { type LoaderFunctionArgs } from 'react-router'
 import { requireUserId } from '#app/utils/auth.server.ts'
 import { NoteEditor } from './__note-editor.tsx'
 
@@ -6,7 +6,7 @@ export { action } from './__note-editor.server.tsx'
 
 export async function loader({ request }: LoaderFunctionArgs) {
 	await requireUserId(request)
-	return json({})
+	return {}
 }
 
 export default NoteEditor
