diff --git a/app/utils/monitoring.client.tsx b/app/utils/monitoring.client.tsx
index afe91b2..587ffaf 100644
--- a/app/utils/monitoring.client.tsx
+++ b/app/utils/monitoring.client.tsx
@@ -1,14 +1,14 @@
-import { useLocation, useMatches } from '@remix-run/react'
+import * as Sentry from '@sentry/react'
+import React from 'react'
 import {
-	init as sentryInit,
-	browserTracingIntegration,
-	replayIntegration,
-	browserProfilingIntegration,
-} from '@sentry/remix'
-import { useEffect } from 'react'
+	createRoutesFromChildren,
+	matchRoutes,
+	useLocation,
+	useNavigationType,
+} from 'react-router'
 
 export function init() {
-	sentryInit({
+	Sentry.init({
 		dsn: ENV.SENTRY_DSN,
 		environment: ENV.MODE,
 		beforeSend(event) {
@@ -25,13 +25,15 @@ export function init() {
 			return event
 		},
 		integrations: [
-			browserTracingIntegration({
-				useEffect,
+			Sentry.replayIntegration(),
+			Sentry.browserProfilingIntegration(),
+			Sentry.reactRouterV7BrowserTracingIntegration({
+				useEffect: React.useEffect,
 				useLocation,
-				useMatches,
+				useNavigationType,
+				createRoutesFromChildren,
+				matchRoutes,
 			}),
-			replayIntegration(),
-			browserProfilingIntegration(),
 		],
 
 		// Set tracesSampleRate to 1.0 to capture 100%
