diff --git a/app/utils/auth.server.ts b/app/utils/auth.server.ts
index 911dfa1..85f917d 100644
--- a/app/utils/auth.server.ts
+++ b/app/utils/auth.server.ts
@@ -1,6 +1,6 @@
 import { type Connection, type Password, type User } from '@prisma/client'
-import { redirect } from '@remix-run/node'
 import bcrypt from 'bcryptjs'
+import { redirect } from 'react-router'
 import { Authenticator } from 'remix-auth'
 import { safeRedirect } from 'remix-utils/safe-redirect'
 import { connectionSessionStorage, providers } from './connections.server.ts'
@@ -53,7 +53,7 @@ export async function requireUserId(
 		redirectTo =
 			redirectTo === null
 				? null
-				: redirectTo ?? `${requestUrl.pathname}${requestUrl.search}`
+				: (redirectTo ?? `${requestUrl.pathname}${requestUrl.search}`)
 		const loginParams = redirectTo ? new URLSearchParams({ redirectTo }) : null
 		const loginRedirect = ['/login', loginParams?.toString()]
 			.filter(Boolean)
