diff --git a/app/routes/users+/$username_+/notes.$noteId_.edit.tsx b/app/routes/users+/$username_+/notes.$noteId_.edit.tsx
index 0d14817..b1ea570 100644
--- a/app/routes/users+/$username_+/notes.$noteId_.edit.tsx
+++ b/app/routes/users+/$username_+/notes.$noteId_.edit.tsx
@@ -1,14 +1,13 @@
 import { invariantResponse } from '@epic-web/invariant'
-import { json, type LoaderFunctionArgs } from '@remix-run/node'
-import { useLoaderData } from '@remix-run/react'
 import { GeneralErrorBoundary } from '#app/components/error-boundary.tsx'
 import { requireUserId } from '#app/utils/auth.server.ts'
 import { prisma } from '#app/utils/db.server.ts'
+import { type Route } from './+types/notes.$noteId_.edit.ts'
 import { NoteEditor } from './__note-editor.tsx'
 
 export { action } from './__note-editor.server.tsx'
 
-export async function loader({ params, request }: LoaderFunctionArgs) {
+export async function loader({ params, request }: Route.LoaderArgs) {
 	const userId = await requireUserId(request)
 	const note = await prisma.note.findFirst({
 		select: {
@@ -28,13 +27,14 @@ export async function loader({ params, request }: LoaderFunctionArgs) {
 		},
 	})
 	invariantResponse(note, 'Not found', { status: 404 })
-	return json({ note: note })
+	return { note }
 }
 
-export default function NoteEdit() {
-	const data = useLoaderData<typeof loader>()
-
-	return <NoteEditor note={data.note} />
+export default function NoteEdit({
+	loaderData,
+	actionData,
+}: Route.ComponentProps) {
+	return <NoteEditor note={loaderData.note} actionData={actionData} />
 }
 
 export function ErrorBoundary() {
