diff --git a/app/routes/users+/$username_+/notes.index.tsx b/app/routes/users+/$username_+/notes.index.tsx
index ab05941..1834c70 100644
--- a/app/routes/users+/$username_+/notes.index.tsx
+++ b/app/routes/users+/$username_+/notes.index.tsx
@@ -1,4 +1,4 @@
-import { type MetaFunction } from '@remix-run/react'
+import { type MetaFunction } from 'react-router'
 import { type loader as notesLoader } from './notes.tsx'
 
 export default function NotesIndexRoute() {
