diff --git a/app/routes/users+/$username.tsx b/app/routes/users+/$username.tsx
index 443133b..98a01d5 100644
--- a/app/routes/users+/$username.tsx
+++ b/app/routes/users+/$username.tsx
@@ -1,6 +1,11 @@
 import { invariantResponse } from '@epic-web/invariant'
-import { json, type LoaderFunctionArgs } from '@remix-run/node'
-import { Form, Link, useLoaderData, type MetaFunction } from '@remix-run/react'
+import {
+	type LoaderFunctionArgs,
+	Form,
+	Link,
+	useLoaderData,
+	type MetaFunction,
+} from 'react-router'
 import { GeneralErrorBoundary } from '#app/components/error-boundary.tsx'
 import { Spacer } from '#app/components/spacer.tsx'
 import { Button } from '#app/components/ui/button.tsx'
@@ -25,7 +30,7 @@ export async function loader({ params }: LoaderFunctionArgs) {
 
 	invariantResponse(user, 'User not found', { status: 404 })
 
-	return json({ user, userJoinedDisplay: user.createdAt.toLocaleDateString() })
+	return { user, userJoinedDisplay: user.createdAt.toLocaleDateString() }
 }
 
 export default function ProfileRoute() {
