diff --git a/app/components/search-bar.tsx b/app/components/search-bar.tsx
index 30c77d3..db7af73 100644
--- a/app/components/search-bar.tsx
+++ b/app/components/search-bar.tsx
@@ -1,5 +1,5 @@
-import { Form, useSearchParams, useSubmit } from '@remix-run/react'
 import { useId } from 'react'
+import { Form, useSearchParams, useSubmit } from 'react-router'
 import { useDebounce, useIsPending } from '#app/utils/misc.tsx'
 import { Icon } from './ui/icon.tsx'
 import { Input } from './ui/input.tsx'
@@ -23,8 +23,8 @@ export function SearchBar({
 		formAction: '/users',
 	})
 
-	const handleFormChange = useDebounce((form: HTMLFormElement) => {
-		submit(form)
+	const handleFormChange = useDebounce(async (form: HTMLFormElement) => {
+		await submit(form)
 	}, 400)
 
 	return (
