diff --git a/app/utils/connections.server.ts b/app/utils/connections.server.ts
index 524c5fa..2d87812 100644
--- a/app/utils/connections.server.ts
+++ b/app/utils/connections.server.ts
@@ -1,4 +1,4 @@
-import { createCookieSessionStorage } from '@remix-run/node'
+import { createCookieSessionStorage } from 'react-router'
 import { type ProviderName } from './connections.tsx'
 import { GitHubProvider } from './providers/github.server.ts'
 import { type AuthProvider } from './providers/provider.ts'
