diff --git a/app/utils/providers/github.server.ts b/app/utils/providers/github.server.ts
index 895271b..054719b 100644
--- a/app/utils/providers/github.server.ts
+++ b/app/utils/providers/github.server.ts
@@ -1,5 +1,5 @@
 import { createId as cuid } from '@paralleldrive/cuid2'
-import { redirect } from '@remix-run/node'
+import { redirect } from 'react-router'
 import { GitHubStrategy } from 'remix-auth-github'
 import { z } from 'zod'
 import { cache, cachified } from '../cache.server.ts'
@@ -38,7 +38,7 @@ export class GitHubProvider implements AuthProvider {
 					throw new Error('Email not found')
 				}
 				const username = profile.displayName
-				const imageUrl = profile.photos[0].value
+				const imageUrl = profile.photos[0]?.value
 				return {
 					email,
 					id: profile.id,
