diff --git a/app/routes/_seo+/sitemap[.]xml.ts b/app/routes/_seo+/sitemap[.]xml.ts
index a3ebb26..5f04d9b 100644
--- a/app/routes/_seo+/sitemap[.]xml.ts
+++ b/app/routes/_seo+/sitemap[.]xml.ts
@@ -1,9 +1,12 @@
 import { generateSitemap } from '@nasa-gcn/remix-seo'
-import { type ServerBuild, type LoaderFunctionArgs } from '@remix-run/node'
+import { type ServerBuild, type LoaderFunctionArgs } from 'react-router'
 import { getDomainUrl } from '#app/utils/misc.tsx'
 
 export async function loader({ request, context }: LoaderFunctionArgs) {
 	const serverBuild = (await context.serverBuild) as { build: ServerBuild }
+
+	// TODO: This is typeerror is coming up since of the remix-run/server-runtime package. We might need to remove/update that one.
+	// @ts-expect-error
 	return generateSitemap(request, serverBuild.build.routes, {
 		siteUrl: getDomainUrl(request),
 		headers: {
