diff --git a/app/routes/resources+/download-user-data.tsx b/app/routes/resources+/download-user-data.tsx
index edd8df4..68c24c3 100644
--- a/app/routes/resources+/download-user-data.tsx
+++ b/app/routes/resources+/download-user-data.tsx
@@ -1,4 +1,4 @@
-import { json, type LoaderFunctionArgs } from '@remix-run/node'
+import { data, type LoaderFunctionArgs } from 'react-router'
 import { requireUserId } from '#app/utils/auth.server.ts'
 import { prisma } from '#app/utils/db.server.ts'
 import { getDomainUrl } from '#app/utils/misc.tsx'
@@ -41,7 +41,7 @@ export async function loader({ request }: LoaderFunctionArgs) {
 
 	const domain = getDomainUrl(request)
 
-	return json({
+	return data({
 		user: {
 			...user,
 			image: user.image
