diff --git a/index.js b/index.js
index bc7c79f..082cd60 100644
--- a/index.js
+++ b/index.js
@@ -1,7 +1,5 @@
 import 'dotenv/config'
 import * as fs from 'node:fs'
-import chalk from 'chalk'
-import closeWithGrace from 'close-with-grace'
 import sourceMapSupport from 'source-map-support'
 
 sourceMapSupport.install({
@@ -18,14 +16,6 @@ sourceMapSupport.install({
 	},
 })
 
-closeWithGrace(async ({ err }) => {
-	if (err) {
-		console.error(chalk.red(err))
-		console.error(chalk.red(err.stack))
-		process.exit(1)
-	}
-})
-
 if (process.env.MOCKS === 'true') {
 	await import('./tests/mocks/index.ts')
 }
