diff --git a/tests/db-utils.ts b/tests/db-utils.ts
index b9bed24..beb412c 100644
--- a/tests/db-utils.ts
+++ b/tests/db-utils.ts
@@ -1,6 +1,5 @@
 import fs from 'node:fs'
 import { faker } from '@faker-js/faker'
-import { type PrismaClient } from '@prisma/client'
 import bcrypt from 'bcryptjs'
 import { UniqueEnforcer } from 'enforce-unique'
 
@@ -15,7 +14,7 @@ export function createUser() {
 			return (
 				faker.string.alphanumeric({ length: 2 }) +
 				'_' +
-				faker.internet.userName({
+				faker.internet.username({
 					firstName: firstName.toLowerCase(),
 					lastName: lastName.toLowerCase(),
 				})
@@ -114,24 +113,3 @@ export async function img({
 		blob: await fs.promises.readFile(filepath),
 	}
 }
-
-export async function cleanupDb(prisma: PrismaClient) {
-	const tables = await prisma.$queryRaw<
-		{ name: string }[]
-	>`SELECT name FROM sqlite_master WHERE type='table' AND name NOT LIKE 'sqlite_%' AND name NOT LIKE '_prisma_migrations';`
-
-	try {
-		// Disable FK constraints to avoid relation conflicts during deletion
-		await prisma.$executeRawUnsafe(`PRAGMA foreign_keys = OFF`)
-		await prisma.$transaction([
-			// Delete all rows from each table, preserving table structures
-			...tables.map(({ name }) =>
-				prisma.$executeRawUnsafe(`DELETE from "${name}"`),
-			),
-		])
-	} catch (error) {
-		console.error('Error cleaning up database:', error)
-	} finally {
-		await prisma.$executeRawUnsafe(`PRAGMA foreign_keys = ON`)
-	}
-}
