diff --git a/tests/mocks/github.ts b/tests/mocks/github.ts
index 3ada535..6290a8f 100644
--- a/tests/mocks/github.ts
+++ b/tests/mocks/github.ts
@@ -54,7 +54,7 @@ function createGitHubUser(code?: string | null) {
 		code,
 		accessToken: `${code}_mock_access_token`,
 		profile: {
-			login: faker.internet.userName(),
+			login: faker.internet.username(),
 			id: faker.string.uuid(),
 			name: faker.person.fullName(),
 			avatar_url: 'https://github.com/ghost.png',
